src/tracer.test.js

```diff
@@ -445,4 +451,52 @@ describe('tracer', () => {
     // No exception.
     expect(handler).toHaveBeenCalledOnce();
   });
+
+  test('performStepFunctionLogic - Happy flow', () => {
+    spies.getRandomId.mockReturnValueOnce('123');
+    spies.isStepFunction.mockReturnValueOnce(true);
+    spies.addStepFunctionEvent.mockImplementationOnce(() => {});
```

Comment:
Try to don't mock your functions

Reply:
The other option is to mock the internals of this function (because it otherwise just doesn't work).
I really believe that it is better to mock the upper level "API" instead of inner implementation...